### PR TITLE
Set app title to "OSCAL Editor" when editing

### PR DESCRIPTION
**Pull request by kylelaker**
This makes it so that when REST Mode is enabled the app is the OSCAL Editor and when it is disabled, it is the OSCAL Viewer. Headers and navigation links are updated accordingly.

### CODE REVIEW

**Comment by kylelaker** [comment]
50/50 on whether I am abusing `useEffect` too much so please tell me if there's a better way

File: example/src/App.js [hunk 1]
```diff
@@ -90,6 +91,13 @@ function App() {
       ReactGA.pageview(location.pathname + location.search);
     }
   }, [location]);
+  useEffect(() => {
+    setAppType(isRestMode ? "Editor" : "Viewer");
+  }, [isRestMode]);
+  useEffect(() => {
+    document.title = `OSCAL ${appType}`;
+  }, [appType]);
+
```
Comment (mikeisen1) [inline example/src/App.js:100]:
```suggestion
useEffect(() => {
    setAppType(isRestMode ? "Editor" : "Viewer");
    document.title = `OSCAL ${appType}`
}, [isRestMode]);
```
Using `useEffect` twice seems unnecessary. Why not do this instead?

**Review by mikeisen1 — CHANGES_REQUESTED** [review]

File: example/src/App.js [hunk 2]
```diff
@@ -90,6 +91,10 @@ function App() {
       ReactGA.pageview(location.pathname + location.search);
     }
   }, [location]);
+useEffect(() => {
+    setAppType(isRestMode ? "Editor" : "Viewer");
+    document.title = `OSCAL ${appType}`
```
Comment (kylelaker) [inline example/src/App.js:96]:
```suggestion
    document.title = `OSCAL ${appType}`;
```

File: example/src/App.js (hunk 1 above)
Comment (kylelaker) [reply example/src/App.js:100]: @mikeisen1 so apparently this isn't quite cool with React. See the message on commit [a2d084f](https://github.com/EasyDynamics/oscal-react-library/pull/398/commits/a2d084fc71579ea000721774b704ad41c53ab6f3).

**Review by mikeisen1 — APPROVED** [review]

**Review by zclarkEDC — APPROVED** [review]

**Comment by rgauss** [comment]
@kylelaker, looks like the End-to-End Cypress tests need to be updated as a part of this.

**Pull request merged**